sdk/resource/detect/env_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package detect
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+func TestDetectOnePair(t *testing.T) {
+	os.Setenv(envVar, "key=value")
+
+	detector := &FromEnv{}
+	res, err := detector.Detect(context.Background())
+	require.NoError(t, err)
+	assert.Equal(t, resource.New(kv.String("key", "value")), res)
+}
+
+func TestDetectEnvTrue(t *testing.T) {
+	os.Setenv("x", "1")
+	os.Setenv(envVar, "key=value, k = v , a=${x:y}, a=${z:y}")
```

Comment:
confused, does this use ${} literally? how does that interrelate to the shell interpreting the $ character?

Reply:
So far what I know about the rule is that value can be specified using the notation ${envVar:defaultValue}, where `envVar` is an environment variable and `defaultValue` is the value to use in case the env var is not set. Refer to https://github.com/open-telemetry/opentelemetry-go/blob/master/exporters/trace/jaeger/env.go#L115. @james-bebbington may have more context.

Reply:
Hmm I guess that is to support nested environment variables (one level). I believe that code you referenced is very specific to Jaeger and don't think you should include it here. There's nothing in the OT spec about this format.